### ISSUE TITLE
Build failure with wlroots12 (1.17)

### ISSUE THREAD

**Issue by lukefromdc** [marcof-nikogo/wf-external-decoration#4]
Using Debian's libwlroots12_0.17.1-2_amd64.deb and wayfire_0.8.0+git20240110-1_amd64.deb on Debian Unstable, I got a build failure in wf-external-decor.cpp on
`if (wlr_surface_is_xwayland_surface(view->get_wlr_surface()))`
which is only ever found at one line in this file so presumably comes in from wlroots. I was able to comment out the offending code with 
```

@@ -840,11 +840,11 @@ class external_decoration_plugin : publi
                     if (deco->decoration->first == 0)
                     {
                         auto view = wf::find_view_for_toplevel(toplevel);
-                        if (wlr_surface_is_xwayland_surface(view->get_wlr_surface()))
+                        /*if (wlr_surface_is_xwayland_surface(view->get_wlr_surface()))
                         {
                             // this is needed for xwayland views, dunno why
                             pending.geometry = wf::expand_geometry_by_margins(pending.geometry, pending.margins);
-                        }
+                        }*/
                         deco->decoration->first = 1;
                     }
                     ev->tx->add_object(deco->decoration);
```
and the build finished and runs. I am playing with this for potential future use in MATE's wayland session.

**Comment by marcof-nikogo** [comment]
Yes, the project was built against wayfire 0.8.x as installed by wf-install script.
Without this check on wlroots 1.17, opening a application using xwayland do you get the decoration properly sized ?

**Comment by lukefromdc** [comment]
No I do not, but I was unable to get the build to finish without commenting that out, so I have to put up with wrong-size xwayland decorations until the an xwayland window has been moved or otherwise forced to re-render once.

**Comment by lukefromdc** [comment]
I found that comenting out only the check for xwayland and applying 
` pending.geometry = wf::expand_geometry_by_margins(pending.geometry, pending.margins);`
not only let the build finish and let xwayland borders render properly, it also fixed some border issues with dialogs that rendered wrong-size decorations momentarily
